public/sw.js

```diff
@@ -1,22 +1,34 @@
 // add arc.io
 importScripts('https://arc.io/arc-sw-core.js')
 
-const VERSION = "1";
+const cacheName = "offline-v1";
+
+const offlineResources = [
+    '/resources/html/offline.html',
+    '/resources/img/bg.webp',
+    '/resources/img/logo_square.svg',
+]
 
 self.addEventListener('install', (event) => {
     event.waitUntil(
-        caches.open(`skyCrypt-offline-v${VERSION}`).then((cache) => {
-            return cache.addAll([
-                '/resources/html/offline.html',
-                '/resources/img/bg.webp',
-                '/resources/img/logo_square.svg',
-            ]);
-        }),
+        caches.has(cacheName).then( async (exists) => {
+            if (!exists) {
+                console.log('reloading')
```

Comment:
is this needed?

Reply:
I'm not sure what part you are asking about so I'll explain what they each do:

`caches.has(cacheName)` this code creates a promise that will return true if a cache with the same name already exists (aka false if we need to update the cache)

`.then( async (exists) => {` this code sets up an async context so that I can use `await` instead of `.then()` and it passes the value from `.has()` into this context as `exists`

`if (!exists) {` this makes it so that we only refetch the offline page when we need to instead of every time we change the SW

`console.log('reloading')` this is a debug line I forgot to remove see 55029d7e19df74e192bf58b1a379f99006fd27d7

Reply:
oops i was trying to quote line 16 but ig it quoted everything